spec.bs

```diff
@@ -289,46 +318,179 @@ To <dfn for="PrivateAggregation">obtain an aggregatable report</dfn> given an [=
+1. Let |sharedInfo| be the result of [=obtaining a report's shared info=] given |report|.
+1. Let |info| be the result of [=UTF-8 encoding=] the [=string/concatenation=] of «"`aggregation_service`", |sharedInfo|».
+1. Let (|kem_id|, |kdf_id|, |aead_id|) be (0x0020, 0x0001, 0x0003).
+
+    Note: These indicate the HPKE algorithm identifiers, specifying the KEM function as DHKEM(X25519, HKDF-SHA256), the KDF function as HKDF-SHA256	and the AEAD function as ChaCha20Poly1305.
+1. Let |hpkeContext| be the result of setting up an [[RFC9180|HPKE]] [[RFC9180#name-encryption-to-a-public-key|sender's context]] with |pkR|, |info|, |kem_id|, |kdf_id| and |aead_id|.
+1. Let |aad| be \`\` (an empty [=byte sequence=]).
+1. Let |encryptedPayload| be the result of [[RFC9180#name-encryption-and-decryption|encrypting]] |plaintext| with |hpkeContext| and |aad|.
+
+To <dfn>obtain the plaintext payload</dfn> given an [=aggregatable report=] |report|, perform the following steps. They return a [=byte sequence=].
+1. Let |payloadData| be a new [=list/is empty|empty=] [=list=].
+1. [=list/iterate|For each=] |contribution| of |report|'s [=aggregatable report/contributions=]:
+    1. Let |contributionData| be an [=ordered map=] of the following key/value pairs:
+
+        : "`bucket`"
+        :: The result of [=encoding an integer for the payload=] given |contribution|["{{PAHistogramContribution/bucket}}"] and 128.
+        : "`value`"
+        :: The result of [=encoding an integer for the payload=] given |contribution|["{{PAHistogramContribution/value}}"] and 32.
+
+    1. [=list/Append=] |contributionData| to |payloadData|.
+1. Let |payload| be an [=ordered map=] of the following key/value pairs:
+
+    : "`data`"
+    :: |payloadData|
+    : "`operation`"
+    :: "`histogram`"
+
+1. Return the [=byte sequence=] resulting from [[!RFC8949|CBOR encoding]] |payload|.
+
+To <dfn>encode an integer for the payload</dfn> given an integer |intToEncode| and an integer |bitLength|, return the representation of |intToEncode| as a big-endian [=byte sequence=] of length |bitLength| / 8, left padding with zeroes as necessary.
+
+To <dfn>obtain a report's shared info</dfn> given an [=aggregatable report=] |report|, perform the following steps. They return a [=string=].
+1. Let |scheduledReportTime| be the [=duration from=] the [=UNIX epoch=] to |report|'s [=aggregatable report/original report time=].
+1. Let |sharedInfo| be an [=ordered map=] of the following key/value pairs:
+
+    : "`api`"
+    :: |report|'s [=aggregatable report/api=]
+    : "`report_id`"
+    :: |report|'s [=aggregatable report/report ID=]
+    : "`reporting_origin`"
+    :: The [=serialization of an origin|serialization=] of |report|'s [=aggregatable report/reporting origin=]
+    : "`scheduled_report_time`"
+    :: The number of seconds in |scheduledReportTime|, rounded down to the nearest number of whole seconds and [=serialize an integer|serialized=]
+    : "`version`"
+    :: "`0.1`"
```

Comment:
We should not hardcode this right?

Reply:
I was thinking we might want to -- if we're updating the version, that means the spec would have changed, right?

Reply:
Hmmm, but does it have to be the same for different implementations? I would slightly lean towards making it less explicit. Wdyt?

Reply:
Yeah, this is an interesting one. We say that obtaining the public key is implementation-defined, which would allow for different aggregation services to be used. But aligning on version should be possible (if one aggregation service needs to make a breaking change, they're presumably already no longer spec compliant). That being said, probably not too critical either way.

@csharrison any thoughts on this?

Reply:
I think speccing the version makes sense to me, if we specify the version as having a 1:1 correspondence with the data format from the client.

Reply:
Thanks! Planning to leave as is then :)